test/ng/directive/inputSpec.js

```diff
@@ -1196,7 +1196,7 @@ describe('input', function() {
 
       it('should validate if max is empty', function() {
         $rootScope.maxVal = undefined;
-        $rootScope.value = new Date(9999, 11, 31, 23, 59, 59);
+        $rootScope.value = new Date(3000, 11, 31, 23, 59, 59);
```

Comment:
I looked a little more into this: MS Edge does not seem to support years > 9999 after all.
([Demo][1]: it's not possible to set the year to > 9999 on Edge)

So, I think this test fix is fine (since `9999` doesn't have any significant value for the test - we just need a later date).

--
Yet, while investigating this, I realized that our internal implementation of date-family inputs, do not support years > 9999.
I will make an issue for that (unless it's so on purpose).

[1]: https://codepen.io/ExpertSystem/pen/YwQWRx?editors=001